### PR TITLE
[test] Add a first test on edge creation

### PR DESCRIPTION
**Pull request by mcharfadi**
# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [x] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+ * Copyright (c) 2024, 2025 Obeo.
```
Comment (frouene) [inline integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts]: wrong data

File: integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        details.getTextField('Name').type('Entity2{enter}');
+        new Explorer().createRepresentation('Root', diagramDescriptionName, diagramTitle);
+        new Diagram().centerViewport();
+      });
+
+      afterEach(() => cy.deleteProject(instanceProjectId));
+
+      it('Then we can create an edge between 2 nodes', () => {
+        const diagram = new Diagram();
+        diagram.getNodes('Simple edges', 'Entity1').click();
```
Comment (frouene) [inline integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts]: Could use `diagramTitle` instead

File: integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        cy.getByTestId('creationhandle-top').should('exist');
+
+        cy.getByTestId('creationhandle-top').trigger('mousedown', { force: true, button: 0 });
+
+        // eslint-disable-next-line cypress/no-unnecessary-waiting
+        cy.wait(500);
+        diagram.getNodes('Simple edges', 'Entity2').trigger('mousemove', { force: true });
```
Comment (frouene) [inline integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts]: Could use `diagramTitle` instead

File: integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+        // eslint-disable-next-line cypress/no-unnecessary-waiting
+        cy.wait(500);
+        diagram.getNodes('Simple edges', 'Entity2').trigger('mouseup', { force: true });
```
Comment (frouene) [inline integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts]: Could use `diagramTitle` instead

Comment (frouene) [inline integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts]: This sequence of instructions could be exposed in a method in Diagram.ts

File: integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts (hunk 2 above)
Comment (frouene) [inline integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts]: There is a method in diagram to select a node

File: packages/sirius-web/backend/sirius-web-e2e-tests/src/main/java/org/eclipse/sirius/web/e2e/tests/diagramEdge/DiagramEdgeDomainProvider.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+import org.eclipse.sirius.components.domain.Entity;
+import org.eclipse.sirius.components.domain.Relation;
+import org.eclipse.sirius.web.application.studio.services.api.IDomainProvider;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+/**
+ * Used to contribute domains for edges.cy.ts tests.
```
Comment (frouene) [inline packages/sirius-web/backend/sirius-web-e2e-tests/src/main/java/org/eclipse/sirius/web/e2e/tests/diagramEdge/DiagramEdgeDomainProvider.java]: for edges-creation.cy.ts

File: packages/sirius-web/backend/sirius-web-e2e-tests/src/main/java/org/eclipse/sirius/web/e2e/tests/diagramEdge/DiagramEdgeViewProvider.java [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+import org.eclipse.sirius.web.e2e.tests.services.SiriusWebE2EColorProvider;
+import org.eclipse.sirius.web.e2e.tests.services.api.IE2EViewProvider;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * The view provider for the edges.cy.ts test suite.
```
Comment (frouene) [inline packages/sirius-web/backend/sirius-web-e2e-tests/src/main/java/org/eclipse/sirius/web/e2e/tests/diagramEdge/DiagramEdgeViewProvider.java]: for the edges-creation.cy.ts

**Review by frouene — CHANGES_REQUESTED** [review]

File: integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts (hunk 1 above)
Comment (mcharfadi) [reply integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts]: done

File: integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts (hunk 2 above)
Comment (mcharfadi) [reply integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts]: done

File: integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts (hunk 3 above)
Comment (mcharfadi) [reply integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts]: done

File: integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts (hunk 4 above)
Comment (mcharfadi) [reply integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts]: done

Comment (mcharfadi) [reply integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts]: done

File: integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts (hunk 2 above)
Comment (mcharfadi) [reply integration-tests/cypress/e2e/project/diagrams/edges-creation.cy.ts]: done

File: packages/sirius-web/backend/sirius-web-e2e-tests/src/main/java/org/eclipse/sirius/web/e2e/tests/diagramEdge/DiagramEdgeDomainProvider.java (hunk 5 above)
Comment (mcharfadi) [reply packages/sirius-web/backend/sirius-web-e2e-tests/src/main/java/org/eclipse/sirius/web/e2e/tests/diagramEdge/DiagramEdgeDomainProvider.java]: done

File: packages/sirius-web/backend/sirius-web-e2e-tests/src/main/java/org/eclipse/sirius/web/e2e/tests/diagramEdge/DiagramEdgeViewProvider.java (hunk 6 above)
Comment (mcharfadi) [reply packages/sirius-web/backend/sirius-web-e2e-tests/src/main/java/org/eclipse/sirius/web/e2e/tests/diagramEdge/DiagramEdgeViewProvider.java]: done

**Review by frouene — APPROVED** [review]
